test/integration/fake_upstream.cc

```diff
@@ -69,16 +69,16 @@ void FakeStream::decodeMetadata(Http::MetadataMapPtr&& metadata_map_ptr) {
 }
 
 void FakeStream::postToConnectionThread(std::function<void()> cb) {
-  parent_.connection().dispatcher().post(cb);
+  parent_.postToConnectionThread(cb);
 }
 
 void FakeStream::encode100ContinueHeaders(const Http::ResponseHeaderMap& headers) {
   std::shared_ptr<Http::ResponseHeaderMap> headers_copy(
       Http::createHeaderMap<Http::ResponseHeaderMapImpl>(headers));
-  parent_.connection().dispatcher().post([this, headers_copy]() -> void {
+  postToConnectionThread([this, headers_copy]() -> void {
     {
       absl::MutexLock lock(&lock_);
-      if (saw_reset_) {
+      if (!parent_.connected() || saw_reset_) {
```

Comment:
optional: might be worth a helper for this given how many places it shows up.

Reply:
Yeah, I can see that as a possible improvement to FakeStream::postToConnectionThread.  I'll look into it as a possible followup.